plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/ScriptGenerationUtil.java

```diff
@@ -368,7 +372,7 @@ public static String instantiateForLoopCountVariables(ForLoopBean forLoopBean, F
 	public static String getForLoopIterateName(ForLoopBean tempForLoopBean, ForLoopBean rootForLoopBean,
 			List<ForLoopBean> forLoopBeanList, Boolean calledFromOutSide) {
 		StringBuilder operationBuilder = new StringBuilder();
-		if (tempForLoopBean.getParentIndex() == -1
+		if (null == rootForLoopBean || tempForLoopBean.getParentIndex() == -1
```

Comment:
Any possibility of "tempForLoopBean" being null here?

Reply:
No, since we do not return null at https://github.com/prabushi/devstudio-tooling-esb/blob/68f5e1dd7fda8d2eb8c432a03fdcca815abbd8b5/plugins/org.wso2.developerstudio.visualdatamapper.diagram/src/org/wso2/developerstudio/datamapper/diagram/custom/util/ScriptGenerationUtil.java#L263